src/convert.rs

```diff
@@ -411,6 +421,28 @@ pub fn elf_to_tbf(
         };
     binary_index += protected_region_size as usize;
 
+    // .crt0_header section is the first content in the TBF after TBF header
+    // the address of this section is used to differentiate from program headers
+    // pointing to another memory location due to alignment.
+    //
+    // This ELF program header mapping issue is observed in (rv32imc):
+    //   examples/tests/bare_bones
+    //   examples/tests/crash_immediately_syscall
+    //
+    // The workaround is to take the crt0_header section address and compare to the first
+    // program header, and then skip the offset differences
+    let crt0_header_address = elf_file
+        .sections
+        .iter()
+        .find_map(|section| {
+            if section.shdr.name == ".crt0_header" {
+                Some(section.shdr.addr as u64)
+            } else {
+                None
+            }
+        })
+        .expect("ELF doesn't have .crt0_header section");
```

Comment:
We don't want this, as we don't want a requirement that there be a section named `.crt0_header`. That is just a convention that libtock-c uses, but not something we want elf2tab to rely on.

Reply:
These are the files that fell into this category:
examples/tests/bare_bones/rv32imc/rv32imc.0x20030880.0x10008000
examples/tests/bare_bones/rv32imc/rv32imc.0x20032080.0x10008000 
examples/tests/bare_bones/rv32imc/rv32imc.0x20034080.0x10008000
examples/tests/bare_bones/rv32imc/rv32imc.0x20030080.0x10005000
examples/tests/crash_immediately_syscall/rv32imc/rv32imc.0x20030880.0x10008000 
examples/tests/crash_immediately_syscall/rv32imc/rv32imc.0x20032080.0x10008000
examples/tests/crash_immediately_syscall/rv32imc/rv32imc.0x20034080.0x10008000
examples/tests/crash_immediately_syscall/rv32imc/rv32imc.0x20030080.0x10005000

This is how one of the program headers look like (others look similar)
Program Headers:
  Type           Offset   VirtAddr   PhysAddr   FileSiz MemSiz  Flg Align
  LOAD           **0x000000** 0x20030000 0x20030000 0x008b0 0x008b0 RWE 0x1000
  LOAD           0x001000 0x10008000 0x200308b0 0x00000 0x00000 RW  0x1000

I can attach these files if you need them.

Reply:
Notably those are the two tests without a `main()` function